File: cpp/56_merge_intervals.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        
        vector<vector<int>> ans;
        for (auto interval : intervals) {
            if (ans.size() == 0) {
                ans.push_back(interval);
            } else {
                auto &last = ans.back();
                if (interval[0] <= last[1]) {
                    last[1] = max(last[1], interval[1]);
                } else {
                    ans.push_back(interval);
                }
            }
        }

        return ans;
    }
};

File: python/56_merge_intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals = sorted(intervals, key=lambda x: x[0])
        ans = [intervals[0]]
        for i in range(1, len(intervals)):
            if intervals[i][0] <= ans[-1][1]:
                ans[-1][1] = max(ans[-1][1], intervals[i][1])
            else:
                ans.append(intervals[i])
        return